core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -465,10 +465,16 @@ static HttpClientConnectionManager configureConnectionManager(Map<String, String
     if (connectionConfig != null) {
       connectionManagerBuilder.setDefaultConnectionConfig(connectionConfig);
     }
+    Integer maxConnections = Integer.getInteger(REST_MAX_CONNECTIONS);
```

Comment:
rather than having the if statement, all of this can be inlined. See my suggestion below

Reply:
updated